### PR TITLE
test: custom spamd in test module for oracle spam/stress tests

### PR DESCRIPTION
**Pull request by jchappelow**
Built on @charithabandi's https://github.com/kwilteam/kwil-db/pull/599

Moved the spam extension files out of the main module and into a `test/nodes/spamd` folder, which creates a customized node as we now suggest to do as per https://github.com/kwilteam/kwil-db/pull/570 with the repo being open sources.  This also builds the node into a different docker image so it doesn't unintentionally become someone's `kwild:latest` or worse, pushed to docker hub.

The test task for this is no longer extension oriented, but specifically for this spam test in the integration test framework.  It simply changes the docker image to use in the `docker-compose.yml.template` execution.

### CODE REVIEW

**Comment by jchappelow** [comment]
Actually spamming now.  Typo in the broadcaster routine fixed.

**Comment by jchappelow** [comment]
`task test:it:spam`

**Comment by jchappelow** [comment]
```
2024-04-02 15:41:05 2024-04-02T20:41:05.505Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.505Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.506Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.506Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.506Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.507Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.507Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.507Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.508Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.508Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.508Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.509Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
2024-04-02 15:41:05 2024-04-02T20:41:05.509Z    info    kwild.tx-router.resolution_spam-resolution      spamd/resoln.go:15      Spam resolution logic approved
```

in logs (for example)

Then when it hits the max number of spams (10,000), you'll see the resolution txns continue to get mined for a while after as they batch in 50

**Review by charithabandi — APPROVED** [review]

File: test/integration/kwild_test.go [hunk 1]
```diff
@@ -503,12 +507,14 @@ func TestKwildEthDepositFundTransfer(t *testing.T) {
 		integration.WithGas(),
 		integration.WithGanache(),
 		integration.WithEthDepositOracle(true),
-		integration.WithConfirmations("0"),
 	}
 
 	testDrivers := strings.Split(*drivers, ",")
 	for _, driverType := range testDrivers {
 		t.Run(driverType+"_driver", func(t *testing.T) {
+			// if driverType == "cli" {
```
[the remark this answers is not in the document]
Comment (jchappelow) [reply test/integration/kwild_test.go]: Ah, I was just searching for this to remove on the wrong branch.  Now removed here: https://github.com/kwilteam/kwil-db/compare/33147fbb069530ecdfa04e6d40ca2e97c17c2e2c..8d4a6f031aa0133cd89f7928cf00912aa02288f1

**Pull request merged**